k8s/cmd/crossconnect-monitor/crossconnect_monitor.go

```diff
@@ -62,16 +61,13 @@ func monitorCrossConnects(address string, continuousMonitor bool) {
 		}
 		println(data)
 
-		prom, err := tools.ReadEnvBool(metricspkg.PrometheusEnv, metricspkg.PrometheusDefault)
-		if err == nil && prom && event.Type == crossconnect.CrossConnectEventType_UPDATE {
-			metrics, metricsIdentifiers, errm := getMetrics(event)
-			if errm != nil {
-				logrus.Infof("failed to get metrics: %v", err)
-			} else {
-				servePrometheus(metricsIdentifiers, metrics)
+		if event.Type == crossconnect.CrossConnectEventType_UPDATE {
+			prom, err := tools.ReadEnvBool(metricspkg.PrometheusEnv, metricspkg.PrometheusDefault)
```

Comment:
Please cache the value fo the env insteadd of reading it at each and evert UPDATE event.